tools/pipelines-tasks/AVDAppAttachPublish/publishAVD.ps1

```diff
@@ -0,0 +1,14 @@
+param([string]$inputJsonStr, [string]$targetDLL)
+
+Write-Output $inputJsonStr
+Write-Output $targetDLL
+
+$scriptDir = $PSScriptRoot
+
+Add-Type -Path $targetDLL
+
+#Create AppAttachKernel object
+$custObj = New-Object AppAttachKernel.AppAttachKernelController($inputJsonStr)
```

Comment:
Does this need to be taking JSON? I would be way more comfortable with passing each parameter separately; the conversion to JSON makes me worry if it is possible to inject some malicious value that would affect other parameters.

Reply:
AppAttach Framework DLLs that we use only takes json as the input. Also, we have a validation module to validate inputs received in this json before performing AVD publish.